google/oauth2/_client_async.py

```diff
@@ -155,20 +173,24 @@ async def jwt_grant(request, token_uri, assertion):
     """
     body = {"assertion": assertion, "grant_type": client._JWT_GRANT_TYPE}
 
-    response_data = await _token_endpoint_request(request, token_uri, body)
+    response_data = await _token_endpoint_request(
+        request, token_uri, body, can_retry=can_retry
+    )
 
     try:
         access_token = response_data["access_token"]
     except KeyError as caught_exc:
-        new_exc = exceptions.RefreshError("No access token in response.", response_data)
+        new_exc = exceptions.RefreshError(
+            "No access token in response.", response_data, retryable=True
```

Comment:
why this is by default Retryable?

Reply:
if it is expected why we expect the retry to succeed?

Reply:
My interpretation was that should this happen we have received a malformed request or error, that ideally should have been handled at the transport layer.

I think the best path is to retry it to see if that resolves the issue. 

If you disagree I am happy to set this to `False`. It is hard to decide what is right :)

Reply:
Let's have them as false since that is the original behavior

We don't want to retry those because - it is not a valid case, we don't expect it often enough to retry just in case